linear_algebra/multivariate_polynomial.lean

```diff
@@ -266,6 +266,40 @@ instance map.is_ring_hom [decidable_eq β] [comm_ring β]
   is_ring_hom (map f : mv_polynomial σ α → mv_polynomial σ β) :=
 map₂.is_ring_hom _ _
 
+variables {p q : mv_polynomial σ α}
+
+section map₂
+
+variables [comm_ring β] [decidable_eq β]
+variables (f : σ → β) (g : α → β) [is_ring_hom g]
+
+@[simp] lemma map₂_sub : (p - q).map₂ f g = p.map₂ f g - q.map₂ f g := is_ring_hom.map_sub _
+
+@[simp] lemma map₂_neg : (-p).map₂ f g = -(p.map₂ f g) := is_ring_hom.map_neg _
+
+end map₂
+
+section map
+
+variables [comm_ring β] [decidable_eq β]
+variables (f : α → β) [is_ring_hom f]
+
+@[simp] lemma map_sub : (p - q).map f = p.map f - q.map f := is_ring_hom.map_sub _
+
+@[simp] lemma map_neg : (-p).map f = -(p.map f) := is_ring_hom.map_neg _
+
+end map
+
+variables (f : σ → α)
+
+@[simp] lemma eval_sub : (p - q).eval f = p.eval f - q.eval f := is_ring_hom.map_sub _
+
+@[simp] lemma eval_neg : (-p).eval f = -(p.eval f) := is_ring_hom.map_neg _
```

Comment:
Can you put `eval.is_ring_hom` just before these two (and similarly for the others) so that they are organized by operation?